src/number_theory/cyclotomic/primitive_roots.lean

```diff
@@ -277,29 +281,113 @@ end
+then the norm of `ζ ^ (p ^ s) - 1` is `p ^ (p ^ s)` if `p ^ (k - s + 1) ≠ 2`. See the next lemmas
+for similar results. -/
+lemma pow_sub_one_norm_prime_pow_ne_two [ne_zero ((p : ℕ) : K)] {k s : ℕ}
   (hζ : is_primitive_root ζ ↑(p ^ (k + 1))) [hpri : fact (p : ℕ).prime]
   [is_cyclotomic_extension {p ^ (k + 1)} K L]
-  (hirr : irreducible (cyclotomic (↑(p ^ (k + 1)) : ℕ) K)) (h : p ≠ 2) :
-  norm K (ζ - 1) = p :=
+  (hirr : irreducible (cyclotomic (↑(p ^ (k + 1)) : ℕ) K))
+  (hirr₁ : irreducible (cyclotomic (↑(p ^ (k - s + 1)) : ℕ) K)) (hs : s ≤ k)
+  (htwo : p ^ (k - s + 1) ≠ 2) : norm K (ζ ^ ((p : ℕ) ^ s) - 1) = p ^ ((p : ℕ) ^ s) :=
 begin
   haveI : ne_zero ((↑(p ^ (k + 1)) : ℕ) : K),
   { refine ⟨λ hzero, _⟩,
-    rw [pow_coe] at hzero,
+    rw [pnat.pow_coe] at hzero,
     simpa [ne_zero.ne ((p : ℕ) : K)] using hzero },
-  have : 2 < p ^ (k + 1),
-  { rw [← coe_lt_coe, pow_coe, pnat.coe_bit0, one_coe],
-    calc 2 < (p : ℕ) : lt_of_le_of_ne hpri.1.two_le (by contrapose! h; exact coe_injective h.symm)
-      ...  = (p : ℕ) ^ 1 : (pow_one _).symm
-      ...  ≤ (p : ℕ) ^ (k + 1) : pow_le_pow (nat.prime.pos hpri.out) le_add_self },
-  simp [sub_one_norm_eq_eval_cyclotomic hζ this hirr]
+  haveI : ne_zero ((↑(p ^ (k - s + 1)) : ℕ) : K),
+  { refine ⟨λ hzero, _⟩,
+    rw [pnat.pow_coe] at hzero,
+    simpa [ne_zero.ne ((p : ℕ) : K)] using hzero },
+
+  let η := ζ ^ ((p : ℕ) ^ s) - 1,
+  let η₁ : K⟮η⟯ := intermediate_field.adjoin_simple.gen K η,
+  have hη : is_primitive_root (η + 1) (p ^ (k + 1 - s)),
+  { rw [sub_add_cancel],
+    refine is_primitive_root.pow (p ^ (k + 1)).pos hζ _,
+    rw [pnat.pow_coe, ← pow_add, add_comm s, nat.sub_add_cancel (le_trans hs (nat.le_succ k))] },
+  haveI : is_cyclotomic_extension {p ^ (k - s + 1)} K K⟮η⟯,
+  { suffices : is_cyclotomic_extension {p ^ (k - s + 1)} K K⟮η + 1⟯.to_subalgebra,
+    { have H : K⟮η + 1⟯.to_subalgebra = K⟮η⟯.to_subalgebra,
+      { simp only [intermediate_field.adjoin_simple_to_subalgebra_of_integral _ _
+          (is_cyclotomic_extension.integral {p ^ (k + 1)} K L _)],
+        refine subalgebra.ext (λ x, ⟨λ hx, adjoin_le _ hx, λ hx, adjoin_le _ hx⟩),
+        { simp only [set.singleton_subset_iff, set_like.mem_coe],
+          exact subalgebra.add_mem _ (subset_adjoin (mem_singleton η)) (subalgebra.one_mem _) },
+        { simp only [set.singleton_subset_iff, set_like.mem_coe],
+          nth_rewrite 0 [← add_sub_cancel η 1],
+          refine subalgebra.sub_mem _ (subset_adjoin (mem_singleton _)) (subalgebra.one_mem _) } },
+      rw [H] at this,
+      exact this },
+    rw [intermediate_field.adjoin_simple_to_subalgebra_of_integral _ _
+      (is_cyclotomic_extension.integral {p ^ (k + 1)} K L _)],
+    have hη' : is_primitive_root (η + 1) ↑(p ^ (k + 1 - s)) := by simpa using hη,
+    convert hη'.adjoin_is_cyclotomic_extension K,
+    rw [nat.sub_add_comm hs] },
+  replace hη : is_primitive_root (η₁ + 1) ↑(p ^ (k - s + 1)),
```

Comment:
you can do the `rw nat.sub_add_comm` earlier, but I think this should just be a general thing for all primitive roots that μ is a primitive root is equivalent to it in an intermediate_field (we have it for units and subrings, iirc)

Reply:
maybe the new `subobject_class` stuff makes it easier

Reply:
What is the lemma about units and subrings? Thanks!

Reply:
I was thinking of [`is_primitive_root.coe_subgroup_iff`](https://leanprover-community.github.io/mathlib_docs/ring_theory/roots_of_unity.html#is_primitive_root.coe_subgroup_iff) and [`coe_units_iff`](https://leanprover-community.github.io/mathlib_docs/ring_theory/roots_of_unity.html#is_primitive_root.coe_units_iff)

Reply:
Good point!

Reply:
This is now done.